src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -419,6 +427,22 @@ def parse_inputs(self, *args, **kwargs) -> TextGenerationInput:
             kwargs["prompt"] = args[0]
             args = args[1:]
 
+        if kwargs:
+            if kwargs.get("generation_kwargs"):
+                generation_kwargs = kwargs.get("generation_kwargs")
+            else:
+                generation_kwargs = {}
+
+            for k, v in kwargs.items():
+                try:
+                    if not generation_kwargs.get(k):
+                        getattr(GenerationDefaults, k)
```

Comment:
Could use hasattr and avoid the try/except?

Reply:
+1, or can also pas in a default to `getattr` to avoid AttributeError

Reply:
oh much better. thanks!